File: src/main/java/cz/wa/secretagent/view/model/EmptyModel.java
package cz.wa.secretagent.view.model;

import java.util.Collections;
import java.util.Set;

import cz.wa.secretagent.view.SAMGraphics;
import cz.wa.secretagent.view.TileId;
import cz.wa.wautils.math.Rectangle2D;
import secretAgent.world.GLModel;

/**
 * Invisible model. 
 * 
 * @author Ondrej Milenovsky
 */
public class EmptyModel implements GLModel {

    public static final EmptyModel INSTANCE = new EmptyModel();

    private EmptyModel() {
    }

    @Override
    public Set<TileId> getAllTileIds() {
        return Collections.emptySet();
    }

    @Override
    public void linkTextures(SAMGraphics graphics) {
        // nothing
    }

    @Override
    public boolean hasLinkedTextures() {
        return true;
    }

    @Override
    public boolean isTransparent() {
        return true;
    }

    @Override
    public Rectangle2D getMaxBounds() {
        return Rectangle2D.ZERO;
    }

    @Override
    public ModelType getType() {
        return ModelType.EMPTY;
    }

}

File: src/main/kotlin/secretAgent/view/model/EmptyModel.kt
package secretAgent.view.model

import cz.wa.wautils.math.Rectangle2D
import secretAgent.view.SamGraphics
import secretAgent.view.renderer.TileId
import secretAgent.world.GLModel
import secretAgent.world.ModelType

/**
 * Invisible model.
 *
 * @author Ondrej Milenovsky
 */
class EmptyModel private constructor() : GLModel {

    override val allTileIds: Set<TileId>
        get() = emptySet()

    override val isTransparent: Boolean
        get() = true

    override val maxBounds: Rectangle2D
        get() = Rectangle2D.ZERO

    override val type: ModelType
        get() = ModelType.EMPTY

    override fun linkTextures(graphics: SamGraphics) {} // nothing

    override fun hasLinkedTextures(): Boolean = true

    companion object {
        @JvmField
        val INSTANCE = EmptyModel()
    }
}